bin/reth/src/providers.rs

```diff
@@ -0,0 +1,136 @@
+use std::path::Path;
+
+use reth_rpc::JwtSecret;
+use tracing::info;
+
+use crate::dirs::{JwtSecretPath, PlatformPath};
+
+type BoxError = Box<dyn std::error::Error>;
```

Comment:
I'd like to have a dedicated Error type that covers all error variants via `thiserror`